force-app/main/default/classes/SOQL.cls

```diff
@@ -270,12 +303,23 @@ public with sharing class SOQL {
      * @return List of contacts with the same Mailing State as the Account Billing State
      */
     public static List<Contact> getAccountsWithContactsByState(Id accountId) {
-        // Query the Account
-
+        // Retrieve the Account with the given ID
+        Account acc = [SELECT Id, BillingState FROM Account WHERE Id = :accountId LIMIT 1];
+    
+        // If no account is found, return an empty list
+        if (acc == null) {
+            return new List<Contact>();
+        }
+    
         // Store the Account's BillingState
-
-        // Query the Contacts
-        return null; // Replace null with the variable you used to store the result
+        String billingState = acc.BillingState;
+    
+        // Query Contacts where MailingState matches the Account's BillingState
+        List<Contact> contactsByState = [SELECT Id, FirstName, LastName, MailingState 
+                                         FROM Contact 
+                                         WHERE AccountId = :accountId AND MailingState = :billingState];
```

Comment:
It looks like the test for this method is failing. See if you can figure out why and update. 🙂